app/src/main/java/com/vero/tasky/auth/util/ValidationResult.kt

```diff
@@ -0,0 +1,8 @@
+package com.vero.tasky.auth.util
+
+import androidx.annotation.StringRes
+
+data class ValidationResult(
```

Comment:
You could use a sealed class to make it easier

```
sealed class ValidationResult {
    object Valid: ValidationResult
    data class Invalid(@StringRes val error: Int): ValidationResult
}
```

Reply:
Makes it a bit cleaner, since you dont need to check if isValid is false or errorMessage != Null

You just check for the type of ValidationResult